addons/escoria-core/ui_library/menus/options/options.gd

```diff
@@ -75,51 +78,75 @@ func _on_language_input(event: InputEvent, language: String):
 # #### Parameters
 # - value: The new volume level
 func _on_sound_volume_changed(value):
-	escoria.settings["sfx_volume"] = value
-	escoria.apply_settings(escoria.settings)
-	settings_changed = true
+	if ESCProjectSettingsManager.get_setting(
+				ESCProjectSettingsManager.SFX_VOLUME
+			) != value:
+		ESCProjectSettingsManager.set_setting(
+			ESCProjectSettingsManager.SFX_VOLUME,
+			value
+		)
+		escoria.settings_manager.apply_settings()
+		settings_changed = true
 
 
 # Music volume was changed
 #
 # #### Parameters
 # - value: The new volume level
 func _on_music_volume_changed(value):
-	escoria.settings["music_volume"] = value
-	escoria.apply_settings(escoria.settings)
-	settings_changed = true
+	if ESCProjectSettingsManager.get_setting(
+				ESCProjectSettingsManager.MUSIC_VOLUME
+			) != value:
+		ESCProjectSettingsManager.set_setting(
+			ESCProjectSettingsManager.MUSIC_VOLUME,
+			value
+		)
+		escoria.settings_manager.apply_settings()
+		settings_changed = true
 
 
 # General volume was changed
 #
 # #### Parameters
 # - value: The new volume level
 func _on_general_volume_changed(value):
-	escoria.settings["master_volume"] = value
-	escoria.apply_settings(escoria.settings)
-	settings_changed = true
+	if ESCProjectSettingsManager.get_setting(
+				ESCProjectSettingsManager.MASTER_VOLUME
+			) != value:
+		ESCProjectSettingsManager.set_setting(
+			ESCProjectSettingsManager.MASTER_VOLUME,
+			value
+		)
+		escoria.settings_manager.apply_settings()
+		settings_changed = true
 
 
 # Speech volume was changed
 #
 # #### Parameters
 # - value: The new volume level
 func _on_speech_volume_value_changed(value: float) -> void:
-	escoria.settings["speech_volume"] = value
-	escoria.apply_settings(escoria.settings)
-	settings_changed = true
+	if ESCProjectSettingsManager.get_setting(
+				ESCProjectSettingsManager.SPEECH_VOLUME
+			) != value:
+		ESCProjectSettingsManager.set_setting(
+			ESCProjectSettingsManager.SPEECH_VOLUME,
+			value
+		)
+		escoria.settings_manager.apply_settings()
+		settings_changed = true
 
 
 # Save the settings
 func _on_apply_pressed():
-	escoria.settings.custom_settings["a_custom_setting"] = 100
-	escoria.save_manager.save_settings()
+	escoria.settings_manager.custom_settings["a_custom_setting"] = 100
```

Comment:
Curious as to what `a_custom_setting` represents?

Also, for the setting keys in this file (like the one above, `speech_volume`, and others), it might be an idea to put them into `const`s.

Reply:
This is an example of the usage of custom settings. It's been there for quite a long time now, since this feature was implemented.
Agreed for `const`, I'll add these.

Reply:
Done